newsfragments/2772.fixed.md

```diff
@@ -0,0 +1 @@
+Fix `PyUnicodeDecodeError_Create` failing unconditionally with a TypeError on PyPy.
```

Comment:
Wouldn't this be a result of undefined behavior?

Reply:
Ah, yes, it's not super clear from the Python docs. It looks like it might be, so I'll amend this in a follow-up PR.